dev/com.ibm.ws.cdi.ejb_fat/fat/src/com/ibm/ws/cdi/ejb/tests/InterceptorsTest.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.cdi.ejb.tests;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.CDIArchiveHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+import com.ibm.websphere.simplicity.beansxml.BeansAsset.DiscoveryMode;
+import com.ibm.ws.cdi.ejb.apps.interceptors.InterceptorsTestServlet;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.annotation.TestServlets;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.rules.repeater.EERepeatActions;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+/**
+ * Testing that interceptors bound with an @Interceptors annotation on an ejb can be resolved
+ * even when they have a common type. In this case we force them not to be CDI beans.
+ */
```

Comment:
I think a bit more detail about the important points of the test scenario would be helpful here.

```suggestion
/**
 * Testing that interceptors bound with an @Interceptors annotation on an ejb can be resolved
 * even when they have a common type.
 * <p>
 * In this test, although the interceptors are beans (due to bean-discovery-mode=all), we must not
 * look them up as beans because the classes are named directly in the {@code @Interceptors}
 * annotation.
 * <p>
 * In particular, type-safe resolution for {@code Interceptor9} would fail because both
 * {@code InterceptorA} and {@code Interceptor9} are beans with that type.
 */
```